sunpy/database/database.py

```diff
@@ -389,7 +398,27 @@ def _download_and_collect_entries(self, query_result, **kwargs):
         if client is None:
             client = VSOClient()
 
-        paths = client.get(query_result, path, methods).wait(progress=progress)
+        remove_list = []
+        delete_entries = []
+        for qr in query_result:
+            temp = tables.DatabaseEntry._from_query_result_block(qr)
+            for database_entry in self:
+                if database_entry.path is not None and temp._compare_attributes(
+                    database_entry, ["source", "provider", "physobs", "fileid",
+                                     "observation_time_start", "observation_time_end",
+                                     "instrument", "size", "wavemin", "wavemax"]):
+                    if overwrite is False:
```

Comment:
rather than having both these lists and the if statement here, why not keep a list of `matched_entries` and then decide what to do with it afterwards based on the state of `overwrite`? I mean `overwrite` is the same for each entry in the loop?